tests/integration_tests/test_integration_elasticsearchstore.py

```diff
@@ -219,3 +180,343 @@ def test_list_all_columns_with_fake_experiment_id(init_store):
+        assert latest_metric == expected_elastic_latest_metrics[i]
+    for i, metric in enumerate(actual_run.metrics):
+        assert metric == expected_elastic_metrics[i]
+
+
+@pytest.mark.usefixtures('init_store')
+def test_log_metric_with_nan_value(init_store):
+    new_metric = Metric(key="nan_metric", value=math.nan, timestamp=10, step=0)
+    expected_elastic_metrics = [ElasticMetric(key="metric0", value=15.0,
+                                              timestamp=1597324762700, step=0, is_nan=False),
+                                ElasticMetric(key="metric0", value=7.0,
+                                              timestamp=1597324762742, step=1, is_nan=False),
+                                ElasticMetric(key="metric0", value=20.0,
+                                              timestamp=1597324762778, step=2, is_nan=False),
+                                ElasticMetric(key="metric1", value=20.0,
+                                              timestamp=1597324762815, step=0, is_nan=False),
+                                ElasticMetric(key="metric1", value=0.0,
+                                              timestamp=1597324762847, step=1, is_nan=False),
+                                ElasticMetric(key="metric1", value=7.0,
+                                              timestamp=1597324762890, step=2, is_nan=False),
+                                ElasticMetric(key="new_metric", value=7.0,
+                                              timestamp=10, step=0, is_nan=False),
+                                ElasticMetric(key="nan_metric", value=0.0,
+                                              timestamp=10, step=0, is_nan=True)]
+    expected_elastic_latest_metrics = [ElasticLatestMetric(key="metric0", value=20.0,
+                                                           timestamp=1597324762778, step=2,
+                                                           is_nan=False),
+                                       ElasticLatestMetric(key="metric1", value=7.0,
+                                                           timestamp=1597324762890, step=2,
+                                                           is_nan=False),
+                                       ElasticLatestMetric(key="new_metric", value=7.0,
+                                                           timestamp=10, step=0, is_nan=False),
+                                       ElasticLatestMetric(key="nan_metric", value=0.0,
+                                                           timestamp=10, step=0, is_nan=True)]
+    init_store.log_metric("7b2e71956f3d4c08b042624a8d83700d", new_metric)
+    actual_run = init_store._get_run("7b2e71956f3d4c08b042624a8d83700d")
+    for i, latest_metric in enumerate(actual_run.latest_metrics):
+        assert latest_metric == expected_elastic_latest_metrics[i]
+    for i, metric in enumerate(actual_run.metrics):
+        assert metric == expected_elastic_metrics[i]
```

Comment:
Same comment than for previous test, only check that you logged the metric and check its value